Fix spelling errors

dforsi opened the pull request:
This PR fixes minor spelling errors, found with an automated tool and some manual editing and review.

I am in doubt about the change of the name s/Dectector/Detector/ in 3 places but it seems a real error.

Found with:
codespell --skip=AUTHORS,CHANGELOG.md --ignore-words-list=ans,ba,fo,hander,hass,hist,nd,proove,te,temperatur,ue --write --interactive 2

zuckschwerdt commented:
Thanks! Changing "Detector" in `.name` is okay, that's documentation only. But please exclude changes to mongoose, it's upstream code.

zuckschwerdt commented:
Oh and klimalogg.c, bresser_6in1.c did get "temperatur" wrong.

dforsi commented:
Removed all */*mongoose* files and fixed 2 "temperatur" (which I had
excluded because of an example in German IN docs/INTEGRATION.md).

Thanks for your feedback!
